test/e2e/tests/confirmations/transactions/shared.ts

```diff
@@ -397,3 +397,23 @@ export async function confirmApproveTransaction(driver: Driver) {
     '.transaction-list__completed-transactions .activity-list-item:nth-of-type(1)',
   );
 }
+
+/**
+ * Creates an ERC20 approve transaction by clicking the approve button on the test dapp
+ *
+ * @param driver - The WebDriver instance
+ */
+export async function createERC20ApproveTransaction(driver: Driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement('#approveTokens');
+}
+
+/**
+ * Waits for the approve transaction confirmation window to open
+ *
+ * @param driver - The WebDriver instance
+ */
+export async function waitForApproveTransactionWindow(driver: Driver) {
+  await driver.waitUntilXWindowHandles(3);
```

Comment:
Can we test if this `waitUntilXWindowHandles(3)` is actually necessary?  If it's not needed, it would be better to remove it.